reactioncommerce/reaction
# 2.0 migrations

Resolves #4630
Impact: **minor**  
Type: **feature**

## Changes
Migrate existing data for cart and orders after an upgrade

## Breaking changes
No, quite the opposite hopefully

## Testing
1. Check out `master`
2. Run Reaction and do complete checkout flow for anonymous and logged in. Also add some more items to carts that you do not check out. With at least one of the carts, do the checkout flow but don't click the final place order button. The goal is to have some carts in various stages of data collection, and some orders for different types of users.
    - It can be good to also enable taxes and add some taxes for your shipping postal code, since some of the conversions are related to taxes.
    - I recommend having at least one order paid for with "example" method and at least one paid for with a credit card through Stripe package.
3. Stop Reaction and check out this branch.
4. In the `Migrations` collection in MongoDB, make sure that the control document has `locked === false` and a version at or less than 38.
5. Run Reaction and verify that you see a message on startup that it ran the migrations up to version 40.
6. Smoke test in the app, including running the existing converted orders through all possible statuses.
